# 알라딘 웹 크롤링
- 가격 , 별점 , 포인트 등등 (정적인 페이지. )
    - requests 패키지로 html를 쉽게 호출했을 때 수집가능 
    - requests.get(url) 코드 한줄로 html 가져올 수 있었다. 
    - 그리고 가져온 html 에서 내가 원하는 태그를 찾았다 (find, find_all)

- 도서 상세페이지에서 **성별연령 정보를 추출하고자 시도했지만 실패**
    -단순 requests로 추출한 html에서는 찾을 수 없었다.


# 알라딘 웹 크롤링
- 가격, 별점, 포인트 등등 (정적인 페이지.)
    - requests 패키지로 html를 쉽게 호출했을때 수집가능
    - requests.get(url) 코드 한줄로 html 가져올 수 있었다.
    - 그리고 가져온 html에서 내가 원하는 태그를 찾았다 (find, find_all)
- 도서 상세페이지에서 **성별연령 정보를 추출하고자 시도했지만 실패.**
    - 단순 requests로 추출한 html에서는 찾을 수 없었다.

In [1]:
# 데이터로드    
import pandas as pd
book = pd.read_csv('2025도서랭킹_추가수집완.csv')

### **그래서, 동적인 페이지를 크롤링 하는 selenium으로 시도해보기로 한다.**
- 사전준비: 패키지, 드라이브 세팅
def START():
START

In [4]:
## 처음 한번만 실행
'''패키지 로드'''
from selenium import webdriver
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.chrome.options import Options
from selenium.webdriver.common.keys import Keys
from selenium.webdriver.common.by import By
from webdriver_manager.chrome import ChromeDriverManager
import time
from bs4 import BeautifulSoup
options = webdriver.ChromeOptions()
##
'''크롬 브라우저 동작방식 설정'''
# options.add_argument('--headless') # 활성화 하면 크롬창 안보임
options.add_argument("--start-maximized") # 브라우저 크기 최대
options.add_experimental_option("detach", True) # 프로그램이 종료되도 창 유지
options.add_argument("--disable-blink-features=AutomationControlled")
# options.add_argument("user-agent=Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/85.0.4183.102 Safari/537.36")
##


In [6]:
#사전준비: 패키지, 드라이브 세팅
def START():...
START

<function __main__.START()>

In [8]:
## url 주소 준비
url = book['d_url'][15]
print(url)

https://www.aladin.co.kr/shop/wproduct.aspx?ItemId=198048203


In [9]:
## 드라이버 생성 및 접속체크
driver = webdriver.Chrome(options=options)
driver.get(url)

### **첫 시도 - 아무 작업이 없이 html 요청 코드는 아래 두줄 고정** 
- driver 에게 html 요청하는 코드는 
- html = driver.page_source
- soup = BeautifulSoup(html, 'html.parser')

In [ ]:
html = driver.page_source
soup = BeautifulSoup(html, 'html.parser')
soup.find('div', class_='buyer_area_a')


<div class="buyer_area_a"><div class="buyer_box"><div class="anay_title"><div class="anay_Tround_L">구매자 분포</div><div class="anay_Tround_S">구매자<br/>분포</div></div><div class="anay_conts"><div class="analysis"><div class="analysis_box"><table border="0" cellpadding="0" cellspacing="0" width="100%"><tbody><tr><td class="tb_woman"><div class="bar" id="bar_w_0" style="width: 8.02%;"><img src="//image.aladin.co.kr/img/shop/2018/analysis_bg1.png"/></div> <div class="per">5.0%</div></td><td class="tb_tit">10대</td><td class="tb_man"><div class="bar" id="bar_m_0" style="width: 0.41%;"><img src="//image.aladin.co.kr/img/shop/2018/analysis_bg1.png"/></div> <div class="per">0.3%</div></td></tr><tr><td class="tb_woman"><div class="bar" id="bar_w_1" style="width: 68.04%;"><img src="//image.aladin.co.kr/img/shop/2018/analysis_bg1.png"/></div> <div class="per">42.8%</div></td><td class="tb_tit">20대</td><td class="tb_man"><div class="bar" id="bar_m_1" style="width: 4.8%;"><img src="//image.aladin.co.kr/i

In [ ]:
# 출력물을 복사 해보자 
print(soup.find('div', class_='buyer_area_a'))

<div class="buyer_area_a"><div class="buyer_box"><div class="anay_title"><div class="anay_Tround_L">구매자 분포</div><div class="anay_Tround_S">구매자<br/>분포</div></div><div class="anay_conts"><div class="analysis"><div class="analysis_box"><table border="0" cellpadding="0" cellspacing="0" width="100%"><tbody><tr><td class="tb_woman"><div class="bar" id="bar_w_0" style="width: 8.02%;"><img src="//image.aladin.co.kr/img/shop/2018/analysis_bg1.png"/></div> <div class="per">5.0%</div></td><td class="tb_tit">10대</td><td class="tb_man"><div class="bar" id="bar_m_0" style="width: 0.41%;"><img src="//image.aladin.co.kr/img/shop/2018/analysis_bg1.png"/></div> <div class="per">0.3%</div></td></tr><tr><td class="tb_woman"><div class="bar" id="bar_w_1" style="width: 68.04%;"><img src="//image.aladin.co.kr/img/shop/2018/analysis_bg1.png"/></div> <div class="per">42.8%</div></td><td class="tb_tit">20대</td><td class="tb_man"><div class="bar" id="bar_m_1" style="width: 4.8%;"><img src="//image.aladin.co.kr/i

<div class="buyer_area_a"><div class="buyer_box"><div class="anay_title"><div class="anay_Tround_L">구매자 분포</div><div class="anay_Tround_S">구매자<br/>분포</div></div><div class="anay_conts"><div class="analysis"><div class="analysis_box"><table border="0" cellpadding="0" cellspacing="0" width="100%"><tbody><tr><td class="tb_woman"><div class="bar" id="bar_w_0" style="width: 8.02%;"><img src="//image.aladin.co.kr/img/shop/2018/analysis_bg1.png"/></div> <div class="per">5.0%</div></td><td class="tb_tit">10대</td><td class="tb_man"><div class="bar" id="bar_m_0" style="width: 0.41%;"><img src="//image.aladin.co.kr/img/shop/2018/analysis_bg1.png"/></div> <div class="per">0.3%</div></td></tr><tr><td class="tb_woman"><div class="bar" id="bar_w_1" style="width: 68.04%;"><img src="//image.aladin.co.kr/img/shop/2018/analysis_bg1.png"/></div> <div class="per">42.8%</div></td><td class="tb_tit">20대</td><td class="tb_man"><div class="bar" id="bar_m_1" style="width: 4.8%;"><img src="//image.aladin.co.kr/img/shop/2018/analysis_bg1.png"/></div> <div class="per">3.0%</div></td></tr><tr><td class="tb_woman"><div class="bar" id="bar_w_2" style="width: 34.6%;"><img src="//image.aladin.co.kr/img/shop/2018/analysis_bg1.png"/></div> <div class="per">21.8%</div></td><td class="tb_tit">30대</td><td class="tb_man"><div class="bar" id="bar_m_2" style="width: 4.23%;"><img src="//image.aladin.co.kr/img/shop/2018/analysis_bg1.png"/></div> <div class="per">2.7%</div></td></tr><tr><td class="tb_woman"><div class="bar" id="bar_w_3" style="width: 21.11%;"><img src="//image.aladin.co.kr/img/shop/2018/analysis_bg1.png"/></div> <div class="per">13.3%</div></td><td class="tb_tit">40대</td><td class="tb_man"><div class="bar" id="bar_m_3" style="width: 4.11%;"><img src="//image.aladin.co.kr/img/shop/2018/analysis_bg1.png"/></div> <div class="per">2.6%</div></td></tr><tr><td class="tb_woman"><div class="bar" id="bar_w_4" style="width: 8.23%;"><img src="//image.aladin.co.kr/img/shop/2018/analysis_bg1.png"/></div> <div class="per">5.2%</div></td><td class="tb_tit">50대</td><td class="tb_man"><div class="bar" id="bar_m_4" style="width: 3.93%;"><img src="//image.aladin.co.kr/img/shop/2018/analysis_bg1.png"/></div> <div class="per">2.5%</div></td></tr><tr><td class="tb_woman"><div class="bar" id="bar_w_5" style="width: 0.73%;"><img src="//image.aladin.co.kr/img/shop/2018/analysis_bg1.png"/></div> <div class="per">0.5%</div></td><td class="tb_tit">60대<img class="Ere_MB_m1" src="https://image.aladin.co.kr/img/shop/2021/over_sixty.png"/></td><td class="tb_man"><div class="bar" id="bar_m_5" style="width: 0.8%;"><img src="//image.aladin.co.kr/img/shop/2018/analysis_bg1.png"/></div> <div class="per">0.5%</div></td></tr><tr><td class="tb_woman_last Ere_PT10 Ere_str">여성</td><td class="tb_tit"></td><td class="tb_man_last Ere_PT10 Ere_str">남성</td></tr></tbody></table></div></div></div> </div></div>

### **과연 스크롤을 굳이 내려야 데이터가 수집될까?**


In [87]:
## 드라이버 생성 및 접속체크
driver = webdriver.Chrome(options=options)
driver.get(url)

html = driver.page_source
soup = BeautifulSoup(html, 'html.parser')
soup.find('div', class_='buyer_area_a')

## print(soup.find('div', class_='buyer_area_a'))

### **굳이** 스크롤 내려야 함   

In [88]:
# n번만 스크롤 내리기
def scroll_n_times(driver, n, interval=0.02):
    body = driver.find_element("tag name", "body")
    for _ in range(n):
        body.send_keys("\ue00f")  # PAGE_DOWN
        time.sleep(interval)
# 사용 예시

scroll_n_times(driver, n=30)


### 스크롤 내렸으니까 html 호출 한번 다시하고 바로 태그도 찾아보자 

In [89]:
html = driver.page_source
soup = BeautifulSoup(html, 'html.parser')
soup.find('div', class_='buyer_area_a')


<div class="buyer_area_a"><div class="buyer_box"><div class="anay_title"><div class="anay_Tround_L">구매자 분포</div><div class="anay_Tround_S">구매자<br/>분포</div></div><div class="anay_conts"><div class="analysis"><div class="analysis_box"><table border="0" cellpadding="0" cellspacing="0" width="100%"><tbody><tr><td class="tb_woman"><div class="bar" id="bar_w_0" style="width: 3.57%;"><img src="//image.aladin.co.kr/img/shop/2018/analysis_bg1.png"/></div> <div class="per">2.2%</div></td><td class="tb_tit">10대</td><td class="tb_man"><div class="bar" id="bar_m_0" style="width: 0.57%;"><img src="//image.aladin.co.kr/img/shop/2018/analysis_bg1.png"/></div> <div class="per">0.4%</div></td></tr><tr><td class="tb_woman"><div class="bar" id="bar_w_1" style="width: 23.19%;"><img src="//image.aladin.co.kr/img/shop/2018/analysis_bg1.png"/></div> <div class="per">14.6%</div></td><td class="tb_tit">20대</td><td class="tb_man"><div class="bar" id="bar_m_1" style="width: 5.59%;"><img src="//image.aladin.co.kr/

<div class="buyer_area_a"><div class="buyer_box"><div class="anay_title"><div class="anay_Tround_L">구매자 분포</div><div class="anay_Tround_S">구매자<br/>분포</div></div><div class="anay_conts"><div class="analysis"><div class="analysis_box"><table border="0" cellpadding="0" cellspacing="0" width="100%"><tbody><tr><td class="tb_woman"><div class="bar" id="bar_w_0" style="width: 8.02%;"><img src="//image.aladin.co.kr/img/shop/2018/analysis_bg1.png"/></div> <div class="per">5.0%</div></td><td class="tb_tit">10대</td><td class="tb_man"><div class="bar" id="bar_m_0" style="width: 0.41%;"><img src="//image.aladin.co.kr/img/shop/2018/analysis_bg1.png"/></div> <div class="per">0.3%</div></td></tr><tr><td class="tb_woman"><div class="bar" id="bar_w_1" style="width: 68.04%;"><img src="//image.aladin.co.kr/img/shop/2018/analysis_bg1.png"/></div> <div class="per">42.8%</div></td><td class="tb_tit">20대</td><td class="tb_man"><div class="bar" id="bar_m_1" style="width: 4.8%;"><img src="//image.aladin.co.kr/img/shop/2018/analysis_bg1.png"/></div> <div class="per">3.0%</div></td></tr><tr><td class="tb_woman"><div class="bar" id="bar_w_2" style="width: 34.6%;"><img src="//image.aladin.co.kr/img/shop/2018/analysis_bg1.png"/></div> <div class="per">21.8%</div></td><td class="tb_tit">30대</td><td class="tb_man"><div class="bar" id="bar_m_2" style="width: 4.23%;"><img src="//image.aladin.co.kr/img/shop/2018/analysis_bg1.png"/></div> <div class="per">2.7%</div></td></tr><tr><td class="tb_woman"><div class="bar" id="bar_w_3" style="width: 21.11%;"><img src="//image.aladin.co.kr/img/shop/2018/analysis_bg1.png"/></div> <div class="per">13.3%</div></td><td class="tb_tit">40대</td><td class="tb_man"><div class="bar" id="bar_m_3" style="width: 4.11%;"><img src="//image.aladin.co.kr/img/shop/2018/analysis_bg1.png"/></div> <div class="per">2.6%</div></td></tr><tr><td class="tb_woman"><div class="bar" id="bar_w_4" style="width: 8.23%;"><img src="//image.aladin.co.kr/img/shop/2018/analysis_bg1.png"/></div> <div class="per">5.2%</div></td><td class="tb_tit">50대</td><td class="tb_man"><div class="bar" id="bar_m_4" style="width: 3.93%;"><img src="//image.aladin.co.kr/img/shop/2018/analysis_bg1.png"/></div> <div class="per">2.5%</div></td></tr><tr><td class="tb_woman"><div class="bar" id="bar_w_5" style="width: 0.73%;"><img src="//image.aladin.co.kr/img/shop/2018/analysis_bg1.png"/></div> <div class="per">0.5%</div></td><td class="tb_tit">60대<img class="Ere_MB_m1" src="https://image.aladin.co.kr/img/shop/2021/over_sixty.png"/></td><td class="tb_man"><div class="bar" id="bar_m_5" style="width: 0.8%;"><img src="//image.aladin.co.kr/img/shop/2018/analysis_bg1.png"/></div> <div class="per">0.5%</div></td></tr><tr><td class="tb_woman_last Ere_PT10 Ere_str">여성</td><td class="tb_tit"></td><td class="tb_man_last Ere_PT10 Ere_str">남성</td></tr></tbody></table></div></div></div> </div></div>


### 수집이 가능하다는 것을 알았습니다! 
[chatbot 에게 명령]
[chatbot에게 명령]
# 내가 테스트하고 성공한 코드
driver = webdriver.Chrome(options=options)
driver.get(url)
scroll_n_times(driver, n=30)
html = driver.page_source
soup = BeautifulSoup(html, 'html.parser')
soup.find('div', 'buyer_area_a')
# soup.find('div', 'buyer_area_a')의 결과물
<div class="buyer_area_a"><div class="buyer_box"><div class="anay_title"><div class="anay_Tround_L">구매자 분포</div><div class="anay_Tround_S">구매자<br/>분포</div></div><div class="anay_conts"><div class="analysis"><div class="analysis_box"><table border="0" cellpadding="0" cellspacing="0" width="100%"><tbody><tr><td class="tb_woman"><div class="bar" id="bar_w_0" style="width: 8.02%;"><img src="//image.aladin.co.kr/img/shop/2018/analysis_bg1.png"/></div> <div class="per">5.0%</div></td><td class="tb_tit">10대</td><td class="tb_man"><div class="bar" id="bar_m_0" style="width: 0.41%;"><img src="//image.aladin.co.kr/img/shop/2018/analysis_bg1.png"/></div> <div class="per">0.3%</div></td></tr><tr><td class="tb_woman"><div class="bar" id="bar_w_1" style="width: 68.04%;"><img src="//image.aladin.co.kr/img/shop/2018/analysis_bg1.png"/></div> <div class="per">42.8%</div></td><td class="tb_tit">20대</td><td class="tb_man"><div class="bar" id="bar_m_1" style="width: 4.8%;"><img src="//image.aladin.co.kr/img/shop/2018/analysis_bg1.png"/></div> <div class="per">3.0%</div></td></tr><tr><td class="tb_woman"><div class="bar" id="bar_w_2" style="width: 34.6%;"><img src="//image.aladin.co.kr/img/shop/2018/analysis_bg1.png"/></div> <div class="per">21.8%</div></td><td class="tb_tit">30대</td><td class="tb_man"><div class="bar" id="bar_m_2" style="width: 4.23%;"><img src="//image.aladin.co.kr/img/shop/2018/analysis_bg1.png"/></div> <div class="per">2.7%</div></td></tr><tr><td class="tb_woman"><div class="bar" id="bar_w_3" style="width: 21.11%;"><img src="//image.aladin.co.kr/img/shop/2018/analysis_bg1.png"/></div> <div class="per">13.3%</div></td><td class="tb_tit">40대</td><td class="tb_man"><div class="bar" id="bar_m_3" style="width: 4.11%;"><img src="//image.aladin.co.kr/img/shop/2018/analysis_bg1.png"/></div> <div class="per">2.6%</div></td></tr><tr><td class="tb_woman"><div class="bar" id="bar_w_4" style="width: 8.23%;"><img src="//image.aladin.co.kr/img/shop/2018/analysis_bg1.png"/></div> <div class="per">5.2%</div></td><td class="tb_tit">50대</td><td class="tb_man"><div class="bar" id="bar_m_4" style="width: 3.93%;"><img src="//image.aladin.co.kr/img/shop/2018/analysis_bg1.png"/></div> <div class="per">2.5%</div></td></tr><tr><td class="tb_woman"><div class="bar" id="bar_w_5" style="width: 0.73%;"><img src="//image.aladin.co.kr/img/shop/2018/analysis_bg1.png"/></div> <div class="per">0.5%</div></td><td class="tb_tit">60대<img class="Ere_MB_m1" src="https://image.aladin.co.kr/img/shop/2021/over_sixty.png"/></td><td class="tb_man"><div class="bar" id="bar_m_5" style="width: 0.8%;"><img src="//image.aladin.co.kr/img/shop/2018/analysis_bg1.png"/></div> <div class="per">0.5%</div></td></tr><tr><td class="tb_woman_last Ere_PT10 Ere_str">여성</td><td class="tb_tit"></td><td class="tb_man_last Ere_PT10 Ere_str">남성</td></tr></tbody></table></div></div></div> </div></div>
# 명령
- 데이터프레임 book에 d_url이 실제 각 페이지의 주소이다.
- 남성에서 가장 높은 %를 보이는 연령대, 여성에서 가장 높은 %를 보이는 연령대를 추출해서 '남성_top_연령대', '여성_top_연령대' 에 추가.
- 첫 3개 페이지만 테스트 해보고 싶다.
- 패키지로드와 드라이버 세팅 코드 필요없음.


In [71]:
book.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4999 entries, 0 to 4998
Data columns (total 21 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   no                4999 non-null   int64  
 1   ranking           4999 non-null   int64  
 2   bookname          4999 non-null   object 
 3   authors           4999 non-null   object 
 4   publisher         4941 non-null   object 
 5   publication_year  4983 non-null   object 
 6   isbn13            4999 non-null   int64  
 7   addition_symbol   4973 non-null   float64
 8   vol               1591 non-null   float64
 9   class_no          4999 non-null   float64
 10  class_nm          4999 non-null   object 
 11  bookImageURL      4870 non-null   object 
 12  bookDtlUrl        4999 non-null   object 
 13  loan_count        4999 non-null   int64  
 14  d_url             4999 non-null   object 
 15  sales_point       4855 non-null   object 
 16  price             4855 non-null   object 


In [90]:
from bs4 import BeautifulSoup

male_top_ages = []
female_top_ages = []

for idx, url in enumerate(book['d_url'][:3]):
    driver.get(url)
    scroll_n_times(driver, n=30)

    soup = BeautifulSoup(driver.page_source, 'html.parser')
    buyer_div = soup.find('div', 'buyer_area_a')

    male_max = {'age': None, 'per': -1}
    female_max = {'age': None, 'per': -1}

    rows = buyer_div.find_all('tr')

    for row in rows:
        age_td = row.find('td', class_='tb_tit')
        if not age_td:
            continue

        age = age_td.get_text(strip=True)

        female_td = row.find('td', class_='tb_woman')
        male_td = row.find('td', class_='tb_man')

        if female_td:
            female_per = female_td.find('div', class_='per')
            if female_per:
                val = float(female_per.text.replace('%', ''))
                if val > female_max['per']:
                    female_max = {'age': age, 'per': val}

        if male_td:
            male_per = male_td.find('div', class_='per')
            if male_per:
                val = float(male_per.text.replace('%', ''))
                if val > male_max['per']:
                    male_max = {'age': age, 'per': val}

    male_top_ages.append(male_max['age'])
    female_top_ages.append(female_max['age'])

book.loc[:2, '남성_top_연령대'] = male_top_ages
book.loc[:2, '여성_top_연령대'] = female_top_ages


In [91]:
result_df = pd.DataFrame({
    'd_url': book.loc[:2, 'd_url'].values,
    '남성_top_연령대': male_top_ages,
    '여성_top_연령대': female_top_ages
})

result_df

d_url 남성_top_연령대 여성_top_연령대
0  https://www.aladin.co.kr/shop/wproduct.aspx?It...        40대        40대
1  https://www.aladin.co.kr/shop/wproduct.aspx?It...        20대        30대
2  https://www.aladin.co.kr/shop/wproduct.aspx?It...        50대        40대

In [92]:
book

no  ranking                      bookname               authors  \
0        1        1              소년이 온다 :한강 장편소설                지은이: 한강   
1        2        2             홍학의 자리 :정해연 장편소설               지은이: 정해연   
2        3        3            작별하지 않는다 :한강 장편소설                지은이: 한강   
3        4        4                 모순 :양귀자 장편소설               지은이: 양귀자   
4        5        5  나는 소망한다 내게 금지된 것을 :양귀자 장편소설               지은이: 양귀자   
...    ...      ...                           ...                   ...   
4994  4996     4995                        까만 토끼   필리파 레더스 글·그림 ;최지현 옮김   
4995  4997     4995                   오싹오싹 좀비 금붕어     글: 모 오하라 ;번역: 지혜연   
4996  4998     4995                        봄이 오리               지은이: 한연진   
4997  4999     4995                  아빠, 나한테 물어봐    버나드 와버 글 ;이수지 그림·옮김   
4998  5000     4995           살인자의 쇼핑몰 :강지영 장편소설               지은이: 강지영   

      publisher publication_year         isbn13  addition_symbol    vol  \
0            창비             2014  9788936434120           3810.0    NaN   
1          문학동네             2021  9788954681155           3810.0    NaN   
2          문학동네             2021  9788954682152           3810.0    NaN   
3            쓰다             2013  9788998441012           3810.0    NaN   
4            쓰다             2019  9788998441074           3810.0    NaN   
...         ...              ...            ...              ...    ...   
4994    알에이치코리아           2013.0  9788925550459          73840.0    NaN   
4995        예림당           2022.0  9788930264914          74840.0    1.0   
4996  보림(보림출판사)           2025.0  9788943317799          77810.0    NaN   
4997        비룡소           2015.0  9788949113609          74800.0  234.0   
4998      자음과모음           2020.0  9788954442145           3810.0    1.0   

      class_no  ... loan_count  \
0       813.62  ...       6300   
1       813.70  ...       5667   
2       813.62  ...       5540   
3       813.62  ...       5478   
4       813.62  ...       4336   
...        ...  ...        ...   
4994    843.00  ...        331   
4995    843.00  ...        331   
4996    813.70  ...        331   
4997    843.00  ...        331   
4998    813.70  ...        331   

                                                  d_url sales_point   price  \
0     https://www.aladin.co.kr/shop/wproduct.aspx?It...     767,918  13,500   
1     https://www.aladin.co.kr/shop/wproduct.aspx?It...      53,420  12,600   
2     https://www.aladin.co.kr/shop/wproduct.aspx?It...     486,356  15,120   
3     https://www.aladin.co.kr/shop/wproduct.aspx?It...     277,357  11,700   
4     https://www.aladin.co.kr/shop/wproduct.aspx?It...      66,025  13,500   
...                                                 ...         ...     ...   
4994  https://www.aladin.co.kr/shop/wproduct.aspx?It...         402   9,000   
4995  https://www.aladin.co.kr/shop/wproduct.aspx?It...         175  10,800   
4996  https://www.aladin.co.kr/shop/wproduct.aspx?It...       1,084  13,500   
4997  https://www.aladin.co.kr/shop/wproduct.aspx?It...       3,761  12,600   
4998  https://www.aladin.co.kr/shop/wproduct.aspx?It...       2,137  12,420   

     page_count   category review_count  average_rating 남성_top_연령대  여성_top_연령대  
0         216.0       한국소설        394.0             9.6        40대         40대  
1         336.0  추리/미스터리소설         53.0             7.3        20대         30대  
2         332.0       한국소설        110.0             9.0        50대         40대  
3         308.0       한국소설        107.0             9.1        nan         nan  
4         368.0       한국소설         38.0             9.4        nan         nan  
...         ...        ...          ...             ...        ...         ...  
4994       40.0        그림책          8.0             9.3        nan         nan  
4995      192.0   동화/명작/고전         14.0            10.0        nan         nan  
4996       32.0        그림책          9.0            10.0        nan         nan  
4997       40.0        그림책         23.

In [94]:
import pandas as pd

book2 = pd.read_csv('2025도서랭킹_최종수집본_.csv')
book2

no  ranking                      bookname               authors  \
0        1        1              소년이 온다 :한강 장편소설                지은이: 한강   
1        2        2             홍학의 자리 :정해연 장편소설               지은이: 정해연   
2        3        3            작별하지 않는다 :한강 장편소설                지은이: 한강   
3        4        4                 모순 :양귀자 장편소설               지은이: 양귀자   
4        5        5  나는 소망한다 내게 금지된 것을 :양귀자 장편소설               지은이: 양귀자   
...    ...      ...                           ...                   ...   
4843  4996     4995                        까만 토끼   필리파 레더스 글·그림 ;최지현 옮김   
4844  4997     4995                   오싹오싹 좀비 금붕어     글: 모 오하라 ;번역: 지혜연   
4845  4998     4995                        봄이 오리               지은이: 한연진   
4846  4999     4995                  아빠, 나한테 물어봐    버나드 와버 글 ;이수지 그림·옮김   
4847  5000     4995           살인자의 쇼핑몰 :강지영 장편소설               지은이: 강지영   

      publisher publication_year         isbn13  addition_symbol    vol  \
0            창비             2014  9788936434120           3810.0    NaN   
1          문학동네             2021  9788954681155           3810.0    NaN   
2          문학동네             2021  9788954682152           3810.0    NaN   
3            쓰다             2013  9788998441012           3810.0    NaN   
4            쓰다             2019  9788998441074           3810.0    NaN   
...         ...              ...            ...              ...    ...   
4843    알에이치코리아           2013.0  9788925550459          73840.0    NaN   
4844        예림당           2022.0  9788930264914          74840.0    1.0   
4845  보림(보림출판사)           2025.0  9788943317799          77810.0    NaN   
4846        비룡소           2015.0  9788949113609          74800.0  234.0   
4847      자음과모음           2020.0  9788954442145           3810.0    1.0   

      class_no  ...   price page_count   category  review_count  \
0       813.62  ...  13,500      216.0       한국소설         394.0   
1       813.70  ...  12,600      336.0  추리/미스터리소설          53.0   
2       813.62  ...  15,120      332.0       한국소설         110.0   
3       813.62  ...  11,700      308.0       한국소설         107.0   
4       813.62  ...  13,500      368.0       한국소설          38.0   
...        ...  ...     ...        ...        ...           ...   
4843    843.00  ...   9,000       40.0        그림책           8.0   
4844    843.00  ...  10,800      192.0   동화/명작/고전          14.0   
4845    813.70  ...  13,500       32.0        그림책           9.0   
4846    843.00  ...  12,600       40.0        그림책          23.0   
4847    813.70  ...  12,420      176.0   액션/스릴러소설          34.0   

     average_rating 최다연령대_여성 최다연령대_남성  성비_여성 20~30대_남성  20~30대_여성  
0               9.6     40.0     40.0    0.7       8.3       31.4  
1               7.3     30.0     20.0    0.8       9.9       48.6  
2               9.0     40.0     50.0    0.7       7.8       25.1  
3               9.1     20.0     30.0    0.8       9.4       54.5  
4               9.4     20.0     20.0    0.9       4.2       76.5  
...             ...      ...      ...    ...       ...        ...  
4843            9.3     30.0     30.0    0.9       6.4       46.2  
4844           10.0     40.0     40.0    0.9       0.9       14.3  
4845           10.0     30.0     30.0    0.8       8.6       30.9  
4846            9.7     30.0     40.0    0.9       4.8       40.3  
4847            8.3     40.0     30.0    0.7      15.0       32.8  

[4848 rows x 26 columns]

In [97]:
pd.set_option('display.max_columns', None)
book2

no  ranking                      bookname               authors  \
0        1        1              소년이 온다 :한강 장편소설                지은이: 한강   
1        2        2             홍학의 자리 :정해연 장편소설               지은이: 정해연   
2        3        3            작별하지 않는다 :한강 장편소설                지은이: 한강   
3        4        4                 모순 :양귀자 장편소설               지은이: 양귀자   
4        5        5  나는 소망한다 내게 금지된 것을 :양귀자 장편소설               지은이: 양귀자   
...    ...      ...                           ...                   ...   
4843  4996     4995                        까만 토끼   필리파 레더스 글·그림 ;최지현 옮김   
4844  4997     4995                   오싹오싹 좀비 금붕어     글: 모 오하라 ;번역: 지혜연   
4845  4998     4995                        봄이 오리               지은이: 한연진   
4846  4999     4995                  아빠, 나한테 물어봐    버나드 와버 글 ;이수지 그림·옮김   
4847  5000     4995           살인자의 쇼핑몰 :강지영 장편소설               지은이: 강지영   

      publisher publication_year         isbn13  addition_symbol    vol  \
0            창비             2014  9788936434120           3810.0    NaN   
1          문학동네             2021  9788954681155           3810.0    NaN   
2          문학동네             2021  9788954682152           3810.0    NaN   
3            쓰다             2013  9788998441012           3810.0    NaN   
4            쓰다             2019  9788998441074           3810.0    NaN   
...         ...              ...            ...              ...    ...   
4843    알에이치코리아           2013.0  9788925550459          73840.0    NaN   
4844        예림당           2022.0  9788930264914          74840.0    1.0   
4845  보림(보림출판사)           2025.0  9788943317799          77810.0    NaN   
4846        비룡소           2015.0  9788949113609          74800.0  234.0   
4847      자음과모음           2020.0  9788954442145           3810.0    1.0   

      class_no        class_nm  \
0       813.62  문학 > 한국문학 > 소설   
1       813.70  문학 > 한국문학 > 소설   
2       813.62  문학 > 한국문학 > 소설   
3       813.62  문학 > 한국문학 > 소설   
4       813.62  문학 > 한국문학 > 소설   
...        ...             ...   
4843    843.00  문학 > 영미문학 > 소설   
4844    843.00  문학 > 영미문학 > 소설   
4845    813.70  문학 > 한국문학 > 소설   
4846    843.00  문학 > 영미문학 > 소설   
4847    813.70  문학 > 한국문학 > 소설   

                                           bookImageURL  \
0     http://image.aladin.co.kr/product/4086/97/cove...   
1     https://image.aladin.co.kr/product/27587/94/co...   
2     https://image.aladin.co.kr/product/27877/5/cov...   
3     https://bookthumb-phinf.pstatic.net/cover/071/...   
4     https://image.aladin.co.kr/product/18936/52/co...   
...                                                 ...   
4843  https://bookthumb-phinf.pstatic.net/cover/072/...   
4844  https://image.aladin.co.kr/product/29086/31/co...   
4845  https://image.aladin.co.kr/product/35938/15/co...   
4846  https://bookthumb-phinf.pstatic.net/cover/096/...   
4847  https://image.aladin.co.kr/product/23178/43/co...   

                                     bookDtlUrl  loan_count  \
0     https://data4library.kr/bookV?seq=2037399        6300   
1     https://data4library.kr/bookV?seq=6050062        5667   
2     https://data4library.kr/bookV?seq=6086446        5540   
3     https://data4library.kr/bookV?seq=1938053        5478   
4     https://data4library.kr/bookV?seq=5395822        4336   
...                                         ...         ...   
4843  https://data4library.kr/bookV?seq=1976287         331   
4844  https://data4library.kr/bookV?seq=6425211         331   
4845  https://data4library.kr/bookV?seq=7021806         331   
4846  https://data4library.kr/bookV?seq=2887808         331   
4847  https://data4library.kr/bookV?seq=5591724         331   

                                                  d_url sales_point   price  \
0     https://www.aladin.co.kr/shop/wproduct.aspx?It...     767,918  13,500   
1     https://www.aladin.co.kr/shop/wproduct.aspx?It...      53,420  12,600   
2     https://www.aladin.co.kr/shop/wproduct.aspx?It...     486,356  15,120   
3     https://w

In [98]:
book.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4999 entries, 0 to 4998
Data columns (total 23 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   no                4999 non-null   int64  
 1   ranking           4999 non-null   int64  
 2   bookname          4999 non-null   object 
 3   authors           4999 non-null   object 
 4   publisher         4941 non-null   object 
 5   publication_year  4983 non-null   object 
 6   isbn13            4999 non-null   int64  
 7   addition_symbol   4973 non-null   float64
 8   vol               1591 non-null   float64
 9   class_no          4999 non-null   float64
 10  class_nm          4999 non-null   object 
 11  bookImageURL      4870 non-null   object 
 12  bookDtlUrl        4999 non-null   object 
 13  loan_count        4999 non-null   int64  
 14  d_url             4999 non-null   object 
 15  sales_point       4855 non-null   object 
 16  price             4855 non-null   object 
